src/interface/blas1_interface.hpp

```diff
@@ -811,14 +811,37 @@ typename sb_handle_t::event_t _rotmg(
   auto d1_view = make_vector_view(_d1, inc, vector_size);
   auto d2_view = make_vector_view(_d2, inc, vector_size);
   auto x1_view = make_vector_view(_x1, inc, vector_size);
-  auto y1_view = make_vector_view(_y1, inc, vector_size);
   auto param_view = make_vector_view(_param, inc, param_size);
 
-  auto operation =
-      Rotmg<decltype(d1_view)>(d1_view, d2_view, x1_view, y1_view, param_view);
-  auto ret = sb_handle.execute(operation, _dependencies);
+  if constexpr (std::is_arithmetic_v<container_3_t>) {
+    constexpr helper::AllocType mem_type = std::is_pointer_v<container_0_t>
+                                               ? helper::AllocType::usm
+                                               : helper::AllocType::buffer;
+    auto _y1_tmp = blas::helper::allocate<mem_type, container_3_t>(
+        1, sb_handle.get_queue());
 
-  return ret;
+    auto copy_y1 = blas::helper::copy_to_device(sb_handle.get_queue(), &_y1,
+                                                _y1_tmp, 1, _dependencies);
+
+    auto y1_view = make_vector_view(_y1_tmp, inc, vector_size);
+    auto operation = Rotmg<decltype(d1_view)>(d1_view, d2_view, x1_view,
+                                              y1_view, param_view);
+
+    auto operator_event =
+        sb_handle.execute(operation, typename sb_handle_t::event_t{copy_y1});
+    if constexpr (mem_type != helper::AllocType::buffer) {
+      // This wait is necessary to free the temporary memory created above and
+      // avoiding the host_task
+      operator_event[0].wait();
+      sycl::free(_y1_tmp, sb_handle.get_queue());
+    }
+    return operator_event;
+  } else {
+    auto y1_view = make_vector_view(_y1, inc, vector_size);
+    auto operation = Rotmg<decltype(d1_view)>(d1_view, d2_view, x1_view,
+                                              y1_view, param_view);
+    return sb_handle.execute(operation, _dependencies);
```

Comment:
Isn't this running rotmg twice for the buffer case?

Reply:
Thank you for your review @Rbiessy 
I am sorry, I am not sure to understand why should it run twice. Rotmg execution is scheduled only once in both code path. The first one is for the new interface, in case `_y1` is a scalar value on the host and the else keep the legacy pointer/buffer interface. Am I missing something? Could you be more specific?

Reply:
From what I see you create a `Rotmg` node and execute before the `if else` and another one in the `else` clause

Reply:
I have a first if-statement to check `_y1` type at line 816. Inside it I have 1 execute at line 831, after handling the allocation and copy of memory. Then a second `if-statement`(line 832) to free usm memory allocation without an `else`. The other` execute` is at line 843 it is part of the `else` matching the `if-statement` at line 816, there aren't other `execute` above line 816.

It is a little bit dense, there are two `ifs` and only one `else`. Maybe I can add a comment to highlight better the code path, would it make it more readable?

Reply:
Oh I see now, that's just me struggling with the GitHub diff and the comment in the middle... Sorry that looks good as it is!